crates/librqbit_core/src/hash_id.rs

```diff
@@ -142,8 +142,42 @@ impl PartialOrd<Id20> for Id20 {
     }
 }
 
+/// A 32-byte hash used in Bittorrent V2, for torrent info hashes, piece hashing, etc.
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub struct Id32(pub [u8; 32]);
```

Comment:
nit: can both of them be implemented with a const generic for the length? Would reduce the source code size as Id20 and Id32 are a copy-paste of each other otherwise.